### ISSUE TITLE
[Help wanted] Strange classification loss behaviour D1 

### ISSUE THREAD

**Issue by landskris** [zylo117/Yet-Another-EfficientDet-Pytorch#638]
Hi, and thanks for this great repo. 
I have been training different EfficientDet models on a custom training set of 3 classes for 1.7k train, ~300 val, ~300 test. 

Firstly, I know the dataset is difficult but I am still surprised as neither D0, D1 or D2 manages to outperform my Faster R-CNN benchmark on the same dataset. ( All EffDet models are about 5 percentage points lower on the COCO main mAP metric) 

However, this question is mainly related to a strange behaviour in the D1 model. 
I have been running identical full fine-tuning settings for D0, D1, and D2. 
_In doing so, I observe that the D1 model's classification loss behaves very strange and spikes during training_. 
The regression losses are not problematic and behaves very similar for all models. 

* Any idea why this happens to D1 specifically? 
* Proposals to handle it? 

Any help is much appreciated! 

Please see the attached training commands and classification losses 

------------ D0 ---------------

`! python train.py -c 0 -p {PROJ_NAME} --head_only False \
 --lr 1e-3 --batch_size 4 --load_weights weights/efficientdet-d0.pth \
 --num_epochs 120 --save_interval 412 \
 --data_path {OFF_LINE_DATA_DIR_BASE} --saved_path {OUTPUT_SAVE_MODEL_DIR} --log_dir {OUTPUT_SAVE_MODEL_DIR}`

![image](https://user-images.githubusercontent.com/43442749/116784364-23f3e400-aa94-11eb-9b54-1878e7141d4c.png)

------------ D1 --------------- 

`python train.py -c 1 -p {PROJ_NAME} --head_only False \
 --lr 1e-3 --batch_size 4 --load_weights weights/efficientdet-d1.pth \
 --num_epochs 120 --save_interval 412 \
 --data_path {OFF_LINE_DATA_DIR_BASE} \
 --saved_path {OUTPUT_SAVE_MODEL_DIR} --log_path {OUTPUT_SAVE_MODEL_DIR}`

Classification Loss ( stopped early due to strange behaviour) 
![image](https://user-images.githubusercontent.com/43442749/116784376-353cf080-aa94-11eb-9343-149b300abb23.png)

------------ D2 ---------------
`! python train.py -c 2 -p {PROJ_NAME} --head_only False \
 --lr 1e-3 --batch_size 4 --load_weights weights/efficientdet-d2.pth\
 --num_epochs 120 --save_interval 412 \
 --data_path {OFF_LINE_DATA_DIR_BASE} --saved_path {OUTPUT_SAVE_MODEL_DIR} \
 --log_path {OUTPUT_SAVE_MODEL_DIR}`

Classification loss ( 2-3 restart of the training which gives the small restart points ) 
![image](https://user-images.githubusercontent.com/43442749/116784392-44bc3980-aa94-11eb-9d02-03b5db20456a.png)

**Comment by zylo117** [comment]
?
Not strange at all.
This is overfitting. You should modify your hyperparameters
